quarkchain/cluster/shard.py

```diff
@@ -299,13 +299,17 @@ def __has_block_hash(self, block_hash):
         return self.shard_state.db.contain_minor_block_by_hash(block_hash)
 
     def __validate_block_headers(self, block_header_list):
-        # TODO: check difficulty and other stuff?
         for i in range(len(block_header_list) - 1):
-            block, prev = block_header_list[i : i + 2]
-            if block.height != prev.height + 1:
+            header, prev = block_header_list[i : i + 2]
+            if header.height != prev.height + 1:
                 return False
-            if block.hash_prev_minor_block != prev.get_hash():
+            if header.hash_prev_minor_block != prev.get_hash():
                 return False
+            shard_id = header.branch.get_shard_id()
+            consensus_type = self.shard.env.quark_chain_config.SHARD_LIST[
+                shard_id
+            ].CONSENSUS_TYPE
+            validate_seal(header, consensus_type)
```

Comment:
do you want to raise exception here or catch it and return False?

Reply:
similar to `master.__validate_block_headers`. whichever way we choose, let's keep them the same